broker/src/main/java/org/apache/rocketmq/broker/processor/AdminBrokerProcessor.java

```diff
@@ -245,7 +245,10 @@ private RemotingCommand updateAndCreateTopic(ChannelHandlerContext ctx,
         topicConfig.setTopicSysFlag(requestHeader.getTopicSysFlag() == null ? 0 : requestHeader.getTopicSysFlag());
 
         this.brokerController.getTopicConfigManager().updateTopicConfig(topicConfig);
-        this.brokerController.registerBrokerAll(false, true, true);
+
+        if (brokerController.getBrokerConfig().getRegisterNameServerPeriod() == 0) {
+            this.brokerController.registerBrokerAll(false, true, true);
```

Comment:
If could not get the topic route info immediately after the topic is created, this may fail some unit tests and confuse some users.

It may be better to detect if the topic route  info is changed, and then decide whether it is necessary to register or not.

Reply:
We may set the interval to 0 to fix unit tests.  Purpose of this pull requests is to group changes and register them all periodically. Users are unlikely to administratively create a topic and send messages immediately. IMO, this change won't bring about much confusion. I won't say no if we keep the default value 0 to maintain behavioral consistency between before and after this pull requests.

Reply:
If creating topic is frequenct, it is better to register the just created topic only. In fact, we do not need to change the register protocol, just do something in preparing heartbeat data.

Reply:
Following your suggestions, name server has to be updated to accept route data delta type of heartbeat. This would force name server upgrade in deployment prior to deployment of brokers having this patch.

Reply:
This results in extra complexities to existing users and deployments.